spec/index.bs

```diff
@@ -2428,114 +2428,115 @@ A {{GPUBindGroupLayout}} object has the following internal slots:
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} visible
+                    on each shader stage in |descriptor|.
+                - There is {{GPULimits/maxDynamicStorageBuffersPerPipelineLayout|GPULimits.maxDynamicStorageBuffersPerPipelineLayout}}
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} with
+                    {{GPUBindGroupLayoutEntry/hasDynamicOffset}} set to `true` visible on any shader
+                    stage in |descriptor|.
+                - There is {{GPULimits/maxSampledTexturesPerShaderStage|GPULimits.maxSampledTexturesPerShaderStage}}
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/sampled-texture}} visible
+                    on each shader stage in |descriptor|.
+                - There is {{GPULimits/maxStorageTexturesPerShaderStage|GPULimits.maxStorageTexturesPerShaderStage}}
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/readonly-storage-texture}}
+                    and {{GPUBindingType/writeonly-storage-texture}} visible on each shader stage in
+                    |descriptor|.
+                - There is {{GPULimits/maxSamplersPerShaderStage|GPULimits.maxSamplersPerShaderStage}}
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/sampler}} visible on each
+                    shader stage in |descriptor|.
+            </div>
 
-        <dfn>dynamic storage buffer validation</dfn>: There must be {{GPULimits/maxDynamicStorageBuffersPerPipelineLayout|GPULimits.maxDynamicStorageBuffersPerPipelineLayout}} or
-                fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} with {{GPUBindGroupLayoutEntry/hasDynamicOffset}} set to `true`
-                in |descriptor| that are visible to any shader stage.
+            Then:
+                1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
+                1. Create a new [=invalid=] {{GPUBindGroupLayout}} and return the result.
 
-        <dfn>sampled texture validation</dfn>: There must be {{GPULimits/maxSampledTexturesPerShaderStage|GPULimits.maxSampledTexturesPerShaderStage}} or
-            fewer |bindingDescriptor|s of type {{GPUBindingType/sampled-texture}} visible on each shader stage in |descriptor|.
+        1. Let |layout| be a new valid {{GPUBindGroupLayout}} object.
+        1. For each {{GPUBindGroupLayoutEntry}} |bindingDescriptor| in |descriptor|.{{GPUBindGroupLayoutDescriptor/entries}}:
+            1. If any of the following conditions are unsatisfied:
+                <div class=validusage>
+                    - If |bindingDescriptor|.{{GPUBindGroupLayoutEntry/visibility}} includes
+                        {{GPUShaderStage/VERTEX}}:
+                        - {{GPUBindingType/storage-buffer}} and {{GPUBindingType/writeonly-storage-texture}}
+                            are not allowed.
```

Comment:
nit: Should say "bindingDescriptor.type is not storage-buffer or writeonly-storage-buffer"